luigi/contrib/s3.py

```diff
@@ -480,9 +480,12 @@ def list(self, path, start_time=None, end_time=None, return_key=False):  # backw
 
     @staticmethod
     def _get_s3_config(key=None):
-        defaults = dict(configuration.get_config().defaults())
+        # LuigiConfigParser has defaults since [DEFAULT] is a special section
+        # LuigiTomlParser does not have it because [DEFAULT] is not a special section
+        luigi_config = configuration.get_config()
+        defaults = dict(luigi_config.defaults()) if hasattr(luigi_config, "defaults") else {}
         try:
-            config = dict(configuration.get_config().items('s3'))
+            config = dict(luigi_config['s3'])
```

Comment:
solution 2: change usage to generic one to use __getitem__ and only take defaults when it's meaningful (ex. LuigiConfigParser)